src/framework/control.ml

```diff
@@ -151,19 +151,21 @@ struct
     let warn_file f = StringMap.iter (warn_func f) in
     if get_bool "dbg.print_dead_code" then (
       if StringMap.is_empty !dead_lines
-      then printf "No lines with dead code found by solver (there might still be dead code removed by CIL).\n" (* TODO https://github.com/goblint/analyzer/issues/94 *)
+      then printf "No lines with dead code found by solver%s.\n" (if GobConfig.get_bool "alldeadcode" then "" else "(there might still be dead code removed by CIL, try running with --enable alldeadcode to see all) ") (* TODO https://github.com/goblint/analyzer/issues/94 *)
```

Comment:
Is there any reason not to always report all dead code? Having to re-run analyses to _maybe_ find more dead code is not very convenient and creates unnecessary confusion between all the different modes, as opposed to just doing the most sensible thing out of the box.

Reply:
I was thinking of program where maybe large parts are under an `if(0)`, and was worried a bit about degrading performance here.

Reply:
I am not sure about the performance impact of the additional impact of dead code on the analyzer either way. Anyways, if we decide that it's worth having this additional option, I would suggest that one adds remarks about the new option in the description string of `dbg.print_dead_code`.  Maybe one can even group the new option with the existing one in the schema.

Reply:
I doubt there's any real-world difference since most programs don't contain a majority of code under `if(0)`. That code is parsed and Cabs is constructed anyway, so the savings are miniscule.

Regarding our dead code options, they should all be refactored to something consistent, e.g. `ana.dead.line`, `ana.dead.branch` and `ana.dead.function`. I also believe that they should all be enabled by default, because forgetting to enable them has hid unsoundness problems for so long. And again, the cost should be insignificant, because they're all basic postprocessing for the parts of code where the expensive transfer functions never even fired.

Also, we should avoid introducing new top-level options, especially for such technical things. Telling the user to try again with `alldeadcode` really has some "try harder" vibes and can potentially be very annoying. Imagine analyzing zstd for over an hour, just to have it essentially say "rerun the whole analysis just to have this extra bit of non-analysis processing that takes an extra fraction of a second". That's just annoying, frustrating and not very user-friendly.

If we really want there to be an option for this, I think we should have it in a new `cil.*` category of options, which essentially just pass through to CIL. It's such an implementation detail, that an user shouldn't have to know if they simply want to know about all of the dead code, as in the spirit of #785 in the first place.

Reply:
I now removed the `alldeadcode` option again.